Lock not connecting to Bluetooth Using YaleAccess Bluetooth integration

### The problem

I have logged this in https://github.com/home-assistant/core/issues/81468 but @bdraco looked over the logs and advised me to log this as an ESPHome isssue.
When using Bluetooth Proxy, my August Lock, using the Yale Access Bluetooth integration, is unable to connect to the ESPHome Bluetooth Proxy.   I have the Proxy less than 4 feet from the lock and still it will not connect.
Logs show connection attempts, but they are failing.
I have included the last logging done.  Hopefully it will be useful

### Which version of ESPHome has the issue?

2022.10.2 - but current DEV version

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.11.1

### What platform are you using?

ESP32

### Board

_No response_

### Component causing the issue

Bluetooth Proxy

### Example YAML snippet

```yaml
substitutions:
  name: "kitchenbleproxy"
packages:
  esphome.bluetooth-proxy: github://esphome/bluetooth-proxies/esp32-generic.yaml@main
esphome:
  name: ${name}
  name_add_mac_suffix: false
logger:
wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
```


### Anything in the logs that might be useful for us?

```txt
2022-11-04 20:37:37.158 DEBUG (MainThread) [yalexs_ble.push] Waiting for advertisement callbacks for FRONT DOOR (78:9C:85:01:CB:3D)
2022-11-04 20:37:42.671 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Connecting scanner version=2, connectable=True
2022-11-04 20:37:43.077 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=0 free=3 limit=3
2022-11-04 20:37:43.240 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.LightState'>, 174803132): LightState(key=174803132, state=True, brightness=0.9019608, color_mode=3, color_brightness=1.0, red=1.0, green=1.0, blue=1.0, white=1.0, color_temperature=0.0, cold_white=1.0, warm_white=1.0, effect='')
2022-11-04 20:37:43.306 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.LightState'>, 2472414331): LightState(key=2472414331, state=True, brightness=0.9019608, color_mode=3, color_brightness=1.0, red=1.0, green=1.0, blue=1.0, white=1.0, color_temperature=0.0, cold_white=1.0, warm_white=1.0, effect='')
2022-11-04 20:37:43.307 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.LightState'>, 3184812465): LightState(key=3184812465, state=False, brightness=1.0, color_mode=3, color_brightness=1.0, red=1.0, green=1.0, blue=1.0, white=1.0, color_temperature=0.0, cold_white=1.0, warm_white=1.0, effect='')
2022-11-04 20:37:43.312 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.LightState'>, 3184812470): LightState(key=3184812470, state=False, brightness=1.0, color_mode=3, color_brightness=1.0, red=1.0, green=1.0, blue=1.0, white=1.0, color_temperature=0.0, cold_white=1.0, warm_white=1.0, effect='')
2022-11-04 20:37:43.326 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.SensorState'>, 1013851870): SensorState(key=1013851870, state=39.0, missing_state=False)
2022-11-04 20:37:43.412 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.SwitchState'>, 320112106): SwitchState(key=320112106, state=False)
2022-11-04 20:38:02.988 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.SensorState'>, 1013851870): SensorState(key=1013851870, state=39.0625, missing_state=False)
2022-11-04 20:38:30.266 DEBUG (MainThread) [homeassistant.components.bluetooth] Triggering bluetooth usb discovery
2022-11-04 20:38:35.368 DEBUG (MainThread) [yalexs_ble.push] Waiting for advertisement callbacks for FRONT DOOR (78:9C:85:01:CB:3D)
2022-11-04 20:38:36.158 DEBUG (MainThread) [homeassistant.components.bluetooth] Rediscovered adapters: {}
2022-11-04 20:39:02.985 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.SensorState'>, 1013851870): SensorState(key=1013851870, state=39.0, missing_state=False)
2022-11-04 20:39:28.557 DEBUG (MainThread) [homeassistant.components.esphome] kitchenbleproxy: 192.168.0.120 disconnected, running disconnected callbacks
2022-11-04 20:39:28.866 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Connecting scanner version=2, connectable=True
2022-11-04 20:39:28.948 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=0 free=3 limit=3
2022-11-04 20:39:29.154 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: F9:C8:D8:80:3C:96 AdvertisementData(local_name='S', manufacturer_data={880: b'\x05\xe4\x00\x00Kitchen0 '}, tx_power=-127, rssi=-80) connectable: True match: set() rssi: -80
2022-11-04 20:39:29.156 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 31:9A:74:40:6C:82 AdvertisementData(manufacturer_data={6: b'\x01\t \x02l\x0f\x8e\x109\t\x08\xe7\xe5\xaf\x99\xa5\xf2\xf2&\xc5\xd55\x96g;N\x06'}, tx_power=-127, rssi=-58) connectable: True match: set() rssi: -58
2022-11-04 20:39:29.255 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: D4:84:57:4B:35:D3 AdvertisementData(local_name='net', manufacturer_data={1704: b'\x01\x01\x002\xd4\x84WK5\xd3'}, service_uuids=['0000ff90-0000-1000-8000-00805f9b34fb', '0000ff80-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-83) connectable: True match: set() rssi: -83
2022-11-04 20:39:29.281 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 51:65:DF:75:88:27 AdvertisementData(manufacturer_data={224: b'\x14\x02\xca\x92n\xad'}, service_data={'0000fe9f-0000-1000-8000-00805f9b34fb': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}, service_uuids=['0000fe9f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-70) connectable: True match: set() rssi: -70
2022-11-04 20:39:29.563 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 4E:4E:4E:7A:8A:45 AdvertisementData(manufacturer_data={224: b'\x00 \xca8\xda\xac'}, service_data={'0000fe9f-0000-1000-8000-00805f9b34fb': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}, service_uuids=['0000fe9f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-61) connectable: True match: set() rssi: -61
2022-11-04 20:39:29.564 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: FC:4F:A2:51:AD:E2 AdvertisementData(local_name='Bathroom globe 2', service_data={'0000fe0f-0000-1000-8000-00805f9b34fb': b'\x1b\x10l\x96\x02'}, service_uuids=['0000fe0f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-82) connectable: True match: set() rssi: -82
2022-11-04 20:39:29.564 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={76: b'\x06\xad\x01\x07\x87\xca\xe2\x86\xaa\x06\x00\x01\x00\x01\x02', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, tx_power=-127, rssi=-76) connectable: True match: {'homekit_controller'} rssi: -76
2022-11-04 20:39:29.564 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Scheduling update to happen in 0.5 seconds
2022-11-04 20:39:29.666 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: F5:FF:FA:CC:57:1F AdvertisementData(local_name='Bathroom globe 1', service_data={'0000fe0f-0000-1000-8000-00805f9b34fb': b'\x1b\x10l\x96\x02'}, service_uuids=['0000fe0f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-86) connectable: True match: set() rssi: -86
2022-11-04 20:39:29.972 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 4D:49:C5:46:9A:9D AdvertisementData(manufacturer_data={224: b'\x00"\xca\x85\x1a\xde'}, service_data={'0000fe9f-0000-1000-8000-00805f9b34fb': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}, service_uuids=['0000fe9f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-74) connectable: True match: set() rssi: -74
2022-11-04 20:39:29.972 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: C4:65:E7:93:81:9B AdvertisementData(local_name='S', manufacturer_data={880: b'\x05\xe4\x00\x00Living room0 '}, tx_power=-127, rssi=-79) connectable: True match: set() rssi: -79
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Update queued
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Queued update starting
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Starting update
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Acquiring lock
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Starting retry loop
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Starting update
2022-11-04 20:39:30.065 DEBUG (MainThread) [yalexs_ble.lock] FRONT DOOR (78:9C:85:01:CB:3D): Connecting to the lock
2022-11-04 20:39:30.065 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 1, last rssi: -76)
2022-11-04 20:39:30.066 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=3 result=True
2022-11-04 20:39:30.106 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:39:30.370 DEBUG (MainThread) [yalexs_ble.push] Canceling in progress update: <Task pending name='Task-15509' coro=<operation_lock.._async_wrap_operation_lock() running at /usr/local/lib/python3.10/site-packages/yalexs_ble/push.py:87> wait_for= cb=[Task.task_wakeup()]>
2022-11-04 20:39:30.373 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): In-progress update canceled due to lock operation or setup timeout
2022-11-04 20:39:30.373 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): In-progress update canceled
2022-11-04 20:39:30.791 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 51:91:91:E6:55:8A AdvertisementData(manufacturer_data={224: b"\x00!\xcalX'"}, service_data={'0000fe9f-0000-1000-8000-00805f9b34fb': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}, service_uuids=['0000fe9f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-88) connectable: True match: set() rssi: -88
2022-11-04 20:39:31.021 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 44:C6:5D:82:DD:3B AdvertisementData(manufacturer_data={76: b'\x10\x05\x01\x14}\xb2\xef'}, tx_power=-127, rssi=-92) connectable: True match: set() rssi: -92
2022-11-04 20:39:31.832 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:31.834 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=0 free=3 limit=3
2022-11-04 20:39:31.846 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=0 free=3 limit=3
2022-11-04 20:39:31.917 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 61:8D:67:AE:22:C7 AdvertisementData(manufacturer_data={224: b'\x00)\xca\xebo`'}, service_data={'0000fe9f-0000-1000-8000-00805f9b34fb': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}, service_uuids=['0000fe9f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-70) connectable: True match: set() rssi: -70
2022-11-04 20:39:31.918 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 05:89:D9:38:B6:F4 AdvertisementData(manufacturer_data={6: b'\x01\t "\xd1\x12\xf5\xfa\xfa\x16\xde\x80H@c\xce\xb9~\t\xc5\xbf(JK\xfeW}'}, tx_power=-127, rssi=-48) connectable: True match: set() rssi: -48
2022-11-04 20:39:31.934 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: FB:69:00:31:AC:56 AdvertisementData(local_name='Bathroom fan', service_data={'0000fe0f-0000-1000-8000-00805f9b34fb': b'Z\x10l\x96\x02'}, service_uuids=['0000fe0f-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-81) connectable: True match: set() rssi: -81
2022-11-04 20:39:32.123 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={465: b'\x00', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, service_uuids=['bd4ac610-0b45-11e3-8ffd-0800200c9a66'], tx_power=-127, rssi=-75) connectable: True match: set() rssi: -75
2022-11-04 20:39:32.561 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: C4:7C:8D:6C:89:E4 AdvertisementData(local_name='Flower care', service_data={'0000fe95-0000-1000-8000-00805f9b34fb': b'q \x98\x00\xdc\xe4\x89l\x8d|\xc4\r\t\x10\x02\x00\x00'}, service_uuids=['0000fe95-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-82) connectable: True match: {'xiaomi_ble'} rssi: -82
2022-11-04 20:39:32.563 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:E4 - C4:7C:8D:6C:89:E4 -> kitchenbleproxy: Connecting (attempt: 1, last rssi: -82)
2022-11-04 20:39:32.563 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=3 result=True
2022-11-04 20:39:32.583 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:39:32.859 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={76: b'\x06\xad\x01\x07\x87\xca\xe2\x86\xaa\x06\x00\x01\x00\x01\x02', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, tx_power=-127, rssi=-76) connectable: True match: set() rssi: -76
2022-11-04 20:39:33.658 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={465: b'\x00', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, service_uuids=['bd4ac610-0b45-11e3-8ffd-0800200c9a66'], tx_power=-127, rssi=-77) connectable: True match: set() rssi: -77
2022-11-04 20:39:34.581 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={76: b'\x06\xad\x01\x07\x87\xca\xe2\x86\xaa\x06\x00\x01\x00\x01\x02', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, tx_power=-127, rssi=-75) connectable: True match: set() rssi: -75
2022-11-04 20:39:34.802 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 80:7A:BF:E9:E4:21 AdvertisementData(local_name='HTC BS 216A96', service_uuids=['0000cb00-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-65) connectable: True match: set() rssi: -65
2022-11-04 20:39:35.008 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 80:7A:BF:E9:E2:5A AdvertisementData(local_name='HTC BS 5A624B', service_uuids=['0000cb00-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-84) connectable: True match: set() rssi: -84
2022-11-04 20:39:38.081 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: C4:7C:8D:6C:89:DC AdvertisementData(local_name='Flower care', service_data={'0000fe95-0000-1000-8000-00805f9b34fb': b'q \x98\x00\xe1\xdc\x89l\x8d|\xc4\r\x08\x10\x01\x02'}, service_uuids=['0000fe95-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-94) connectable: True match: {'xiaomi_ble'} rssi: -94
2022-11-04 20:39:38.083 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:DC - C4:7C:8D:6C:89:DC -> kitchenbleproxy: Connecting (attempt: 1, last rssi: -94)
2022-11-04 20:39:38.084 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:39:38.095 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:39:38.573 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={465: b'\x00', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, service_uuids=['bd4ac610-0b45-11e3-8ffd-0800200c9a66'], tx_power=-127, rssi=-76) connectable: True match: set() rssi: -76
2022-11-04 20:39:39.700 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:39.701 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:39:39.701 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:39:39.701 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:E4 - C4:7C:8D:6C:89:E4 -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 1, last rssi: -82)
2022-11-04 20:39:40.252 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:E4 - C4:7C:8D:6C:89:E4 -> kitchenbleproxy: Connecting (attempt: 2, last rssi: -82)
2022-11-04 20:39:40.252 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:39:40.333 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:39:40.372 DEBUG (MainThread) [yalexs_ble.push] Waiting for advertisement callbacks for FRONT DOOR (78:9C:85:01:CB:3D)
2022-11-04 20:39:40.372 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Scheduling update to happen in 0.5 seconds
2022-11-04 20:39:40.873 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Update queued
2022-11-04 20:39:40.873 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Queued update starting
2022-11-04 20:39:40.873 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Starting update
2022-11-04 20:39:40.873 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Acquiring lock
2022-11-04 20:39:40.873 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Starting retry loop
2022-11-04 20:39:40.874 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Starting update
2022-11-04 20:39:40.874 DEBUG (MainThread) [yalexs_ble.lock] FRONT DOOR (78:9C:85:01:CB:3D): Connecting to the lock
2022-11-04 20:39:40.874 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 1, last rssi: -76)
2022-11-04 20:39:40.874 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=1 result=True
2022-11-04 20:39:40.945 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=3 free=0 limit=3
2022-11-04 20:39:43.302 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:43.303 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:43.303 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:39:43.303 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 1, last rssi: -76)
2022-11-04 20:39:43.309 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:39:43.633 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: C4:7C:8D:6C:89:E4 AdvertisementData(local_name='Flower care', service_data={'0000fe95-0000-1000-8000-00805f9b34fb': b'q \x98\x00\xdd\xe4\x89l\x8d|\xc4\r\x04\x10\x02\xd3\x00'}, service_uuids=['0000fe95-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-87) connectable: True match: set() rssi: -87
2022-11-04 20:39:43.854 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 2, last rssi: -76)
2022-11-04 20:39:43.854 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=1 result=True
2022-11-04 20:39:43.951 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=3 free=0 limit=3
2022-11-04 20:39:51.188 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Connection state changed to connected=True mtu=23 error=0
2022-11-04 20:39:51.188 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Connection state changed to connected=True mtu=23 error=0
2022-11-04 20:39:51.188 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: connected, registering for disconnected callbacks
2022-11-04 20:39:51.188 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=3 free=0 limit=3
2022-11-04 20:39:51.189 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Cached services miss
2022-11-04 20:39:51.392 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Got services: ESPHomeBluetoothGATTServices(address=216039227689444, services=[BluetoothGATTService(uuid='00001800-0000-1000-8000-00805f9b34fb', handle=1, characteristics=[BluetoothGATTCharacteristic(uuid='00002a00-0000-1000-8000-00805f9b34fb', handle=3, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00002a01-0000-1000-8000-00805f9b34fb', handle=5, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00002a02-0000-1000-8000-00805f9b34fb', handle=7, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='00002a04-0000-1000-8000-00805f9b34fb', handle=9, properties=2, descriptors=[])]), BluetoothGATTService(uuid='00001801-0000-1000-8000-00805f9b34fb', handle=12, characteristics=[BluetoothGATTCharacteristic(uuid='00002a05-0000-1000-8000-00805f9b34fb', handle=14, properties=34, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=15)])]), BluetoothGATTService(uuid='0000fe95-0000-1000-8000-00805f9b34fb', handle=16, characteristics=[BluetoothGATTCharacteristic(uuid='00000001-0000-1000-8000-00805f9b34fb', handle=18, properties=24, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=19)]), BluetoothGATTCharacteristic(uuid='00000002-0000-1000-8000-00805f9b34fb', handle=21, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00000004-0000-1000-8000-00805f9b34fb', handle=23, properties=18, descriptors=[]), BluetoothGATTCharacteristic(uuid='00000007-0000-1000-8000-00805f9b34fb', handle=25, properties=8, descriptors=[]), BluetoothGATTCharacteristic(uuid='00000010-0000-1000-8000-00805f9b34fb', handle=27, properties=8, descriptors=[]), BluetoothGATTCharacteristic(uuid='00000013-0000-1000-8000-00805f9b34fb', handle=29, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='00000014-0000-1000-8000-00805f9b34fb', handle=31, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00001001-0000-1000-8000-00805f9b34fb', handle=33, properties=16, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=34)])]), BluetoothGATTService(uuid='0000fef5-0000-1000-8000-00805f9b34fb', handle=35, characteristics=[BluetoothGATTCharacteristic(uuid='8082caa8-41a6-4021-91c6-56f9b954cc34', handle=37, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='724249f0-5ec3-4b5f-8804-42345af08651', handle=39, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='6c53db25-47a1-45fe-a022-7c92fb334fd4', handle=41, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='9d84b9a3-000c-49d8-9183-855b673fda31', handle=43, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='457871e8-d516-4ca1-9116-57d0b17b9cb2', handle=45, properties=14, descriptors=[]), BluetoothGATTCharacteristic(uuid='5f78df94-798c-46f5-990a-b3eb6a065c88', handle=47, properties=18, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=48)])]), BluetoothGATTService(uuid='00001204-0000-1000-8000-00805f9b34fb', handle=49, characteristics=[BluetoothGATTCharacteristic(uuid='00001a00-0000-1000-8000-00805f9b34fb', handle=51, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='00001a01-0000-1000-8000-00805f9b34fb', handle=53, properties=26, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=54)]), BluetoothGATTCharacteristic(uuid='00001a02-0000-1000-8000-00805f9b34fb', handle=56, properties=2, descriptors=[BluetoothGATTDescriptor(uuid='00001a02-0000-1000-8000-00805f9b34fb', handle=57)])]), BluetoothGATTService(uuid='00001206-0000-1000-8000-00805f9b34fb', handle=58, characteristics=[BluetoothGATTCharacteristic(uuid='00001a11-0000-1000-8000-00805f9b34fb', handle=60, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00001a10-0000-1000-8000-00805f9b34fb', handle=62, properties=26, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=63)]), BluetoothGATTCharacteristic(uuid='00001a12-0000-1000-8000-00805f9b34fb', handle=65, properties=2, descriptors=[BluetoothGATTDescriptor(uuid='00001a12-0000-1000-8000-00805f9b34fb', handle=66)])])])
2022-11-04 20:39:51.394 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Cached services saved
2022-11-04 20:39:51.394 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:E4 - C4:7C:8D:6C:89:E4 -> kitchenbleproxy: Connected (attempt: 2, last rssi: -82)
2022-11-04 20:39:51.522 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Out of connection slots, waiting for a free one
2022-11-04 20:39:51.579 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: Connection state changed to connected=False mtu=0 error=22
2022-11-04 20:39:51.579 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: Flower care - C4:7C:8D:6C:89:E4: BLE device disconnected
2022-11-04 20:39:51.579 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:39:55.338 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:DC - C4:7C:8D:6C:89:DC -> kitchenbleproxy: Timed out trying to connect (attempt: 1, last rssi: -94)
2022-11-04 20:39:55.589 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:DC - C4:7C:8D:6C:89:DC -> kitchenbleproxy: Connecting (attempt: 2, last rssi: -94)
2022-11-04 20:39:55.589 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=1 result=True
2022-11-04 20:39:55.619 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:39:56.517 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 46:24:89:13:CE:70 AdvertisementData(manufacturer_data={76: b'\x10\x06k\x1d\tW\xa8\x80'}, tx_power=-127, rssi=-96) connectable: True match: set() rssi: -96
2022-11-04 20:39:58.951 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:58.951 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:58.951 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:39:58.952 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 2, last rssi: -76)
2022-11-04 20:39:58.952 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:39:58.953 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:39:59.505 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 3, last rssi: -76)
2022-11-04 20:39:59.505 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:39:59.566 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:40:01.837 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={76: b'\x06\xad\x01\x07\x87\xca\xe2\x86\xaa\x06\x00\x01\x00\x01\x02', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, tx_power=-127, rssi=-76) connectable: True match: set() rssi: -76
2022-11-04 20:40:01.838 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Scheduling update to happen in 0.5 seconds
2022-11-04 20:40:02.339 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Rescheduling update since one already in progress
2022-11-04 20:40:02.340 DEBUG (MainThread) [yalexs_ble.push] FRONT DOOR (78:9C:85:01:CB:3D): Scheduling update to happen in 29.5 seconds
2022-11-04 20:40:02.741 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:02.742 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:02.742 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:02.743 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:02.744 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:02.745 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 3, last rssi: -76)
2022-11-04 20:40:02.991 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenledstrip: dispatching update with key (<class 'aioesphomeapi.model.SensorState'>, 1013851870): SensorState(key=1013851870, state=39.0625, missing_state=False)
2022-11-04 20:40:03.296 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 4, last rssi: -76)
2022-11-04 20:40:03.297 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:40:03.457 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:40:04.481 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:04.481 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:04.481 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:04.482 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:04.482 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 4, last rssi: -76)
2022-11-04 20:40:04.483 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:05.034 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 5, last rssi: -76)
2022-11-04 20:40:05.034 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:40:05.121 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:40:05.404 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={465: b'\x00', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, service_uuids=['bd4ac610-0b45-11e3-8ffd-0800200c9a66'], tx_power=-127, rssi=-76) connectable: True match: set() rssi: -76
2022-11-04 20:40:06.037 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:06.037 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:06.037 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:06.038 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:06.038 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:06.038 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 5, last rssi: -76)
2022-11-04 20:40:06.241 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: 78:9C:85:01:CB:3D AdvertisementData(local_name='AugSW38', manufacturer_data={76: b'\x06\xad\x01\x07\x87\xca\xe2\x86\xaa\x06\x00\x01\x00\x01\x02', 22: b',\x0e\xa5\xcc\xc9\xd5[\xc1D\x8d\x8aE<\x90\x14u\xd5k\x00\x00\x00\x00\x00'}, tx_power=-127, rssi=-75) connectable: True match: set() rssi: -75
2022-11-04 20:40:06.589 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 6, last rssi: -75)
2022-11-04 20:40:06.589 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:40:06.650 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:40:07.575 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: C4:7C:8D:6C:89:E4 AdvertisementData(local_name='Flower care', service_data={'0000fe95-0000-1000-8000-00805f9b34fb': b'q \x98\x00\xdf\xe4\x89l\x8d|\xc4\r\x08\x10\x01\x00'}, service_uuids=['0000fe95-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-82) connectable: True match: set() rssi: -82
2022-11-04 20:40:09.089 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:09.089 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:09.089 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:09.089 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:09.090 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 6, last rssi: -75)
2022-11-04 20:40:09.090 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:09.641 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 7, last rssi: -75)
2022-11-04 20:40:09.641 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:40:09.703 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:40:11.561 DEBUG (MainThread) [homeassistant.components.bluetooth.manager] kitchenbleproxy: C4:7C:8D:6C:89:E4 AdvertisementData(local_name='Flower care', service_data={'0000fe95-0000-1000-8000-00805f9b34fb': b'q \x98\x00\xe0\xe4\x89l\x8d|\xc4\r\t\x10\x02\x00\x00'}, service_uuids=['0000fe95-0000-1000-8000-00805f9b34fb'], tx_power=-127, rssi=-82) connectable: True match: set() rssi: -82
2022-11-04 20:40:12.842 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:DC - C4:7C:8D:6C:89:DC -> kitchenbleproxy: Timed out trying to connect (attempt: 2, last rssi: -94)
2022-11-04 20:40:13.094 DEBUG (MainThread) [bleak_retry_connector] C4:7C:8D:6C:89:DC - C4:7C:8D:6C:89:DC -> kitchenbleproxy: Connecting (attempt: 3, last rssi: -94)
2022-11-04 20:40:13.094 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=1 result=True
2022-11-04 20:40:13.261 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
2022-11-04 20:40:16.463 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:16.463 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:16.464 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] kitchenbleproxy: AugSW38 - 78:9C:85:01:CB:3D: Connection state changed to connected=False mtu=0 error=62
2022-11-04 20:40:16.464 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:16.465 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=1 free=2 limit=3
2022-11-04 20:40:16.466 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Failed to connect: Error ESP_GATT_CONN_FAIL_ESTABLISH while connecting: Connection failed to establish, backing off: 0.55 (attempt: 7, last rssi: -75)
2022-11-04 20:40:17.018 DEBUG (MainThread) [bleak_retry_connector] FRONT DOOR (78:9C:85:01:CB:3D) - 78:9C:85:01:CB:3D -> kitchenbleproxy: Connecting (attempt: 8, last rssi: -75)
2022-11-04 20:40:17.019 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] kitchenbleproxy: Checking can connect, available=True, ble_connections_free=2 result=True
2022-11-04 20:40:17.122 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] kitchenbleproxy: BLE connection limits: used=2 free=1 limit=3
```


### Additional information

The above log was the latest one that @bdraco looked at and determined I should likely log an issue with ESPHome team.

Not sure if I did this correctly, but the config YAML at least validated. :)

Note that I'm up-to-date with ESPHome v2022.11.2. I had previously updated ("Update All") my device configs before trying to take some time today to try the bleeding edge.

I added this to the config for one of my ESP32s:

```
external_components:
  - source: github://bdraco/esphome@ble_debug_services
    components: [ esp32_ble_client, esp32_ble_tracker, bluetooth_proxy ]
```

It validated — after multiple previous attempts to do the same thing didn't — so I then tried to install it wirelessly. I got the following:

```
src/esphome/components/esp32_ble_client/ble_characteristic.cpp: In member function 'esphome::esp32_ble_client::BLEDescriptor* esphome::esp32_ble_client::BLECharacteristic::get_descriptor(esphome::esp32_ble_tracker::ESPBTUUID)':
src/esphome/components/esp32_ble_client/ble_characteristic.cpp:53:6: error: expected '(' before 'unlikely'
   if unlikely (!this->parsed)
      ^
src/esphome/components/esp32_ble_client/ble_characteristic.cpp: In member function 'esphome::esp32_ble_client::BLEDescriptor* esphome::esp32_ble_client::BLECharacteristic::get_descriptor_by_handle(uint16_t)':
src/esphome/components/esp32_ble_client/ble_characteristic.cpp:65:6: error: expected '(' before 'unlikely'
   if unlikely (!this->parsed)
      ^
*** [/data/firebeetle-061cb8/.pioenvs/firebeetle-061cb8/src/esphome/components/esp32_ble_client/ble_characteristic.cpp.o] Error 1
src/esphome/components/bluetooth_proxy/bluetooth_proxy.cpp: In member function 'virtual void esphome::bluetooth_proxy::BluetoothProxy::loop()':
src/esphome/components/bluetooth_proxy/bluetooth_proxy.cpp:115:10: error: expected '(' before 'likely'
       if likely (!service->parsed)
          ^
src/esphome/components/bluetooth_proxy/bluetooth_proxy.cpp:122:12: error: expected '(' before 'likely'
         if likely (!characteristic->parsed)
            ^
*** [/data/firebeetle-061cb8/.pioenvs/firebeetle-061cb8/src/esphome/components/bluetooth_proxy/bluetooth_proxy.cpp.o] Error 1
```

--- a/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp
+++ b/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp
@@ -166,9 +166,9 @@ void ESP32BLETracker::loop() {
   ClientStateCounts counts = this->count_client_states_();
   if (counts != this->client_state_counts_) {
     this->client_state_counts_ = counts;
-    ESP_LOGD(TAG, "connecting: %d, discovered: %d, disconnecting: %d, active: %d", this->client_state_counts_.connecting,
-             this->client_state_counts_.discovered, this->client_state_counts_.disconnecting,
-             this->client_state_counts_.active);
+    ESP_LOGD(TAG, "connecting: %d, discovered: %d, disconnecting: %d, active: %d",
+             this->client_state_counts_.connecting, this->client_state_counts_.discovered,
+             this->client_state_counts_.disconnecting, this->client_state_counts_.active);
   }
 
   // Scanner failure: reached when set_scanner_state_(FAILED) or scan_set_param_failed_ set

--- a/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp
+++ b/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp
@@ -191,13 +191,15 @@ void ESP32BLETracker::loop() {
   */
 
   // Start scan: reached when scanner_state_ becomes IDLE (via set_scanner_state_()) and
-  // all clients are idle (their state changes increment version when they finish)
+  // no clients are in the transient CONNECTING / DISCOVERED / DISCONNECTING states
+  // (their state changes increment version when they finish). CONNECTED / ESTABLISHED
+  // clients do NOT block this branch — the coex revert below has its own active-count gate.
   if (this->scanner_state_ == ScannerState::IDLE && !counts.connecting && !counts.disconnecting && !counts.discovered) {
 #ifdef USE_ESP32_BLE_SOFTWARE_COEXISTENCE
     // Only revert to BALANCE when no connections are active. Established connections
-    // continue to need PREFER_BT so the lock's GATT Write Response can reach us while
-    // WiFi traffic (advertisement upload, log streaming) competes for the shared radio.
-    // Reverting too early causes Bluedroid to time out at ~20s and synthesize status=133.
+    // continue to need PREFER_BT so peer GATT responses can reach us while WiFi traffic
+    // (advertisement upload, log streaming) competes for the shared radio. Reverting too
+    // early causes Bluedroid to time out at ~20s and synthesize status=133.
     if (!counts.active) {
       this->update_coex_preference_(false);
     }

--- a/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp
+++ b/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp
@@ -166,8 +166,9 @@ void ESP32BLETracker::loop() {
   ClientStateCounts counts = this->count_client_states_();
   if (counts != this->client_state_counts_) {
     this->client_state_counts_ = counts;
-    ESP_LOGD(TAG, "connecting: %d, discovered: %d, disconnecting: %d", this->client_state_counts_.connecting,
-             this->client_state_counts_.discovered, this->client_state_counts_.disconnecting);
+    ESP_LOGD(TAG, "connecting: %d, discovered: %d, disconnecting: %d, active: %d", this->client_state_counts_.connecting,
+             this->client_state_counts_.discovered, this->client_state_counts_.disconnecting,
+             this->client_state_counts_.active);
   }
 
   // Scanner failure: reached when set_scanner_state_(FAILED) or scan_set_param_failed_ set
@@ -193,7 +194,13 @@ void ESP32BLETracker::loop() {
   // all clients are idle (their state changes increment version when they finish)
   if (this->scanner_state_ == ScannerState::IDLE && !counts.connecting && !counts.disconnecting && !counts.discovered) {
 #ifdef USE_ESP32_BLE_SOFTWARE_COEXISTENCE
-    this->update_coex_preference_(false);
+    // Only revert to BALANCE when no connections are active. Established connections
+    // continue to need PREFER_BT so the lock's GATT Write Response can reach us while
+    // WiFi traffic (advertisement upload, log streaming) competes for the shared radio.
+    // Reverting too early causes Bluedroid to time out at ~20s and synthesize status=133.
+    if (!counts.active) {
+      this->update_coex_preference_(false);
+    }
 #endif
     if (this->scan_continuous_) {
       this->start_scan_(false);  // first = false
@@ -701,9 +708,10 @@ void ESP32BLETracker::dump_config() {
                 this->scan_active_ ? "ACTIVE" : "PASSIVE", YESNO(this->scan_continuous_));
   ESP_LOGCONFIG(TAG,
                 "  Scanner State: %s\n"
-                "  Connecting: %d, discovered: %d, disconnecting: %d",
+                "  Connecting: %d, discovered: %d, disconnecting: %d, active: %d",
                 this->scanner_state_to_string_(this->scanner_state_), this->client_state_counts_.connecting,
-                this->client_state_counts_.discovered, this->client_state_counts_.disconnecting);
+                this->client_state_counts_.discovered, this->client_state_counts_.disconnecting,
+                this->client_state_counts_.active);
   if (this->scan_start_fail_count_) {
     ESP_LOGCONFIG(TAG, "  Scan Start Fail Count: %d", this->scan_start_fail_count_);
   }

--- a/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h
+++ b/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h
@@ -160,9 +160,13 @@ struct ClientStateCounts {
   uint8_t connecting = 0;
   uint8_t discovered = 0;
   uint8_t disconnecting = 0;
+  // CONNECTED + ESTABLISHED clients. Tracked so coex stays at PREFER_BT
+  // while active connections may still need to send/receive GATT traffic.
+  uint8_t active = 0;
 
   bool operator==(const ClientStateCounts &other) const {
-    return connecting == other.connecting && discovered == other.discovered && disconnecting == other.disconnecting;
+    return connecting == other.connecting && discovered == other.discovered && disconnecting == other.disconnecting &&
+           active == other.active;
   }
 
   bool operator!=(const ClientStateCounts &other) const { return !(*this == other); }
@@ -380,6 +384,10 @@ class ESP32BLETracker : public Component,
           break;
         case ClientState::CONNECTING:
           counts.connecting++;
+          break;
+        case ClientState::CONNECTED:
+        case ClientState::ESTABLISHED:
+          counts.active++;
           break;
         default:
           break;

